runelite-client/src/main/java/net/runelite/client/plugins/bosslogger/BossLoggerPlugin.java

```diff
@@ -0,0 +1,1365 @@
+		if (bossLoggerConfig.showChatMessages())
+		{
+			final ChatMessageBuilder chatMessage = new ChatMessageBuilder()
+					.append("<col=" + messageColor + ">")
+					.append(message)
+					.append("</col>");
+
+			chatMessageManager.queue(QueuedMessage.builder()
+					.type(ChatMessageType.EXAMINE_ITEM)
+					.runeLiteFormattedMessage(chatMessage.build())
+					.build());
+		}
+
+		if (bossLoggerConfig.showTrayAlerts())
+		{
+			notifier.notify(message);
+		}
+	}
+
+	void loadTabData(Tab tab)
+	{
+		loadLootEntries(tab);
+	}
+
+	// Load data for all bosses being recorded
+	void loadAllData()
+	{
+		for (Tab tab : Tab.values())
+		{
+			if (isBeingRecorded(tab.getName()))
+			{
+				loadLootEntries(tab);
+			}
+		}
+	}
+
+	// Returns stored data by tab name
+	ArrayList<LootEntry> getData(String type)
+	{
+		// Loot Entries are stored on lootMap by boss name (upper cased)
+		String name = Tab.getByName(type).getBossName().toUpperCase();
+		return lootMap.get(name);
+	}
+
+	//
+	// LootEntry helper functions
+	//
+
+	// Adds the data to the correct boss log file
+	private void AddBossLootEntry(String bossName, ArrayList<DropEntry> drops)
+	{
+		int KC = killcountMap.get(bossName.toUpperCase());
+		LootEntry newEntry = new LootEntry(KC, drops);
+		addLootEntry(bossName, newEntry);
+		BossLoggedAlert(bossName + " kill added to log.");
+	}
+
+	// Create Loot Entry for ItemContainer
+	private LootEntry createLootEntry(int kill_count, ItemContainer container)
+	{
+		ArrayList<DropEntry> drops = new ArrayList<>();
+		for (Item item : container.getItems())
+		{
+			int id = item.getId();
+			int amount = item.getQuantity();
+			drops.add(new DropEntry(id, amount));
+		}
+
+		return new LootEntry(kill_count, drops);
+	}
+
+	// Add Loot Entry to the necessary file
+	private void addLootEntry(String bossName, LootEntry entry)
+	{
+		// Update data inside plugin
+		ArrayList<LootEntry> loots = lootMap.get(bossName.toUpperCase());
+		loots.add(entry);
+		lootMap.put(bossName.toUpperCase(), loots);
+		// Convert entry to JSON
+		String dataAsString = RuneLiteAPI.GSON.toJson(entry);
+		// Grab file by username or loots directory if not logged in
+		updatePlayerFolder();
+		String fileName = filenameMap.get(bossName.toUpperCase());
+		// Open File and append data
+		File lootFile = new File(playerFolder, fileName);
+		try
+		{
+			BufferedWriter file = new BufferedWriter(new FileWriter(String.valueOf(lootFile), true));
+			file.append(dataAsString);
+			file.newLine();
+			file.close();
+		}
+		catch (IOException ioe)
+		{
+			log.warn("Error writing loot data in file.", ioe);
+		}
+
+		// Update tab if being displayed;
+		Tab tab = Tab.getByBossName(bossName);
+		if (isBeingRecorded(tab.getName()))
+		{
+			panel.updateTab(tab.getName());
+		}
+	}
+
+	private synchronized void clearLootFile(Tab tab)
+	{
+		String fileName = filenameMap.get(tab.getBossName().toUpperCase());
+		File lootFile = new File(playerFolder, fileName);
+
+		try
```

Comment:
Same as the other try, if you're using resources make sure to close them in a finally, or try with